crates/sc-proof-of-time/src/lib.rs

```diff
@@ -72,7 +71,7 @@ pub async fn start_slot_worker<Block, Client, SC, Worker, SO, CIDP>(
 
     let mut maybe_last_proven_slot = None;
 
-    while let Some(PotSlotInfo { slot, checkpoints }) = slot_info_stream.next().await {
+    while let Ok(PotSlotInfo { slot, checkpoints }) = slot_info_stream.recv().await {
```

Comment:
This is not a correct refactoring because of the semantics of full broadcast channel. When one of the receiver fills up, old messages will be dropped, but it doesn't mean that the stream is already over, see https://docs.rs/tokio/latest/tokio/sync/broadcast/index.html#lagging

Refactoring is likely similarly incorrect for domains as well.